tools/importer-rest-api-specs/components/parser/dataworkarounds/workaround_alertsmanagement.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package dataworkarounds
+
+import (
+	"fmt"
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+
+	importerModels "github.com/hashicorp/pandora/tools/importer-rest-api-specs/models"
+)
+
+var _ workaround = workaroundAlertsManagement{}
+
+// workaroundAlertsManagement works around the missing discriminator implementations for
+// ActionRuleProperties. This workaround can be removed when v4.0 of the AzureRM Provider
+// has been released.
+type workaroundAlertsManagement struct {
+}
+
+func (workaroundAlertsManagement) IsApplicable(apiDefinition *importerModels.AzureApiDefinition) bool {
+	serviceMatches := apiDefinition.ServiceName == "AlertsManagement"
+	apiVersionMatches := apiDefinition.ApiVersion == "2019-05-05-preview"
+	return serviceMatches && apiVersionMatches
+}
+
+func (workaroundAlertsManagement) Name() string {
+	return "AlertsManagement"
+}
+
+func (workaroundAlertsManagement) Process(apiDefinition importerModels.AzureApiDefinition) (*importerModels.AzureApiDefinition, error) {
+	resource, ok := apiDefinition.Resources["ActionRules"]
+	if !ok {
+		return nil, fmt.Errorf("expected a Resource named `ActionRules`")
+	}
```

Comment:
it's unlikely this changes, but since we're assuming it exists below we should probably validate it here all the same:

```suggestion
	}
	if _, ok := resource.Models["ActionRuleProperties"]; !ok {
		return nil, fmt.Errorf("expected the APIResource `ActionRules` to contain a model named `ActionRuleProperties` but didn't find it")
	}
```